### ISSUE TITLE
can-compile doesn't work

### ISSUE THREAD

**Issue by cesarin3134** [daffl/can-compile#31]
I used canjs 1.1.8 version to develop a web app, used ejs(embed javascript) to handle the views templates. now i want to build all files .ejs with can-compile, i read the documentation here http://daffl.github.io/can-compile/#command-line but i doesn't work.

this is the content of my ejs file to make a test :

    <ul class="bookmarkOptionsUl">
    <li class="bookmarkoptions modifica">Modify</li>
    <li class="bookmarkoptions elimina">Delete</li>

i run all command line wrote in documentation but nothing.

in this case i run

can-compile submenuBookmark.ejs --can 1.1.8 
Compiling bookmark/views/bookmark.ejs

i get this error message : There was an error: undefined

**Comment by daffl** [comment]
It seems to work fine for me. Are you using Windows?

**Comment by cesarin3134** [comment]
i'm using a OSX YOSEMITE.

**Comment by daffl** [comment]
I did (with Node 0.10.36):

```
npm install can-compile -g
echo "<ul class=\"bookmarkOptionsUl\"><li class=\"bookmarkoptions modifica\">Modify</li><li class=\"bookmarkoptions elimina\">Delete</li>" > test.ejs
can-compile test.ejs --can 1.1.8
```

And got

```
Compiling test.ejs
Wrote compiled views to views.production.js
```

You could try changing https://github.com/daffl/can-compile/blob/master/bin/can-compile#L30 to `console.error(err)` to see what the actual error looks like. Or try using the Grunt or Gulp task to get a better error stacktrace.

**Comment by cesarin3134** [comment]
i updated nodejs to version v0.12.0 and tried reinstall can-compile but i got this error  
    npm WARN engine can-compile@0.7.1: wanted: {"node":"~0.10"} (current:        
    {"node":"0.12.0","npm":"2.5.1"}) 

and then it remains in pending.

**Comment by cesarin3134** [comment]
ok,  i installed node 0.10.36V, install can-compile 
run : 
   sudo npm install can-compile -g
    echo "<ul class=\"bookmarkOptionsUl\"><li class=\"bookmarkoptions modifica\">Modify</li><li      class=\"bookmarkoptions elimina\">Delete</li>" > test.ejs
  can-compile test.ejs --can 1.1.8
and i get this error : 
    Compiling test.ejs
    There was an error: undefined

**Comment by daffl** [comment]
Are you behind a proxy? Node needs to be able to have internet access (see #30). Since I can't reproduce the issue you will have to do some debugging yourself. You can clone the repository then `npm install` and try running `bin/can-compile view.ejs --can 1.1.8`.

The error gets reported in https://github.com/daffl/can-compile/blob/master/bin/can-compile#L28 so you can try logging what gets actually thrown (since `err.message` seems to be undefined).

**Comment by cesarin3134** [comment]
i tried the same step on other mac osx maverick , and it work for me , but would like to understand why it doesn't work for me on yosemite OSX. 
the environment is the same : 
    node -v : v0.10.29
    can-compile@0.7.0
thank a lot

**Comment by daffl** [comment]
Cesar, like I said before I need a little more information than the environment and the Node version (specifically what the error is) otherwise there is not much I can help with.

**Comment by cesarin3134** [comment]
this is my file ejs
![file1](https://cloud.githubusercontent.com/assets/1257506/6252386/3fb468b6-b79a-11e4-88b8-18772365c79e.png)
my terminal : 
![terminal](https://cloud.githubusercontent.com/assets/1257506/6252611/e4154492-b79b-11e4-9e83-a6df1e857d71.png)

**Comment by cesarin3134** [comment]
if you need anything else let me know. thank

**Comment by daffl** [comment]
Yes please. I need you to modify the can-compile code and `console.log(err)` in https://github.com/daffl/can-compile/blob/master/bin/can-compile#L34 instead of `err.message` so that we know what the actual error is.

**Comment by cesarin3134** [comment]
ok i do it : 
![modified_can_compile](https://cloud.githubusercontent.com/assets/1257506/6254037/dcd41794-b7a5-11e4-92b3-f40aef93daf2.png)
i got this error : 
![newerror](https://cloud.githubusercontent.com/assets/1257506/6254051/fa7e9d0a-b7a5-11e4-96df-de608465ec54.png)

**Comment by daffl** [comment]
That error means that the client was not able to connect to the web (see http://stackoverflow.com/questions/17690803/node-js-getaddrinfo-enotfound). Try running a Node script like

```js
var http = require("http");
http.get('http://canjs.com/release/latest/can.jquery.js', function (res) {
    var data = "";

    res.on("data", function (chunk) {
        data += chunk;
    });

    res.on("end", function () {
        console.log(data);
    });
});
```

And check if you are getting the same error.

**Comment by cesarin3134** [comment]
ok i run the node script and it work, i run too the can-compile command in another place out of the office with another network too and it work, now i know the the problem is the node proxy setting into the office it doesn't work , but i don't know how to solve it , do you have an idea??
thank a lot for all the support.

**Comment by daffl** [comment]
I'm not sure this is going to work. I found a related issue that prevents JSDOM from loading scripts through a proxy at https://github.com/tmpvar/jsdom/issues/567. It seems to be using `http_proxy` environment variable though so try setting that.

**Comment by cesarin3134** [comment]
you are right , i run the command in my work but it doesn't work. 
i set node proxy in this mode : 
    `npm config set proxy http://myproxy:8080`
    `npm config set https-proxy http://myproxy:8080`
    `npm config set http-proxy http://myproxy:8080`

but nothing, i'm trying to solve the problem. thanks

**Comment by daffl** [comment]
We might be able to solve this issue by addressing #30 and make the tests work offline. I'm not entirely sure what the best solution for that would be though.

**Comment by daffl** [comment]
I think this one can also be solved via #36 by providing your own paths to jQuery can CanJS.
